src/sentry/auth/superuser.py

```diff
@@ -275,12 +292,54 @@ def set_logged_in(self, user, current_datetime=None):
         request = self.request
         if current_datetime is None:
             current_datetime = timezone.now()
+
+        token = get_random_string(12)
+
+        if is_self_hosted() or not self._need_validation():
```

Comment:
if we have an option in settings that is default off, i don't think you need to check `is_self_hosted` here.